modules/secrets-store-aws.adoc

```diff
@@ -0,0 +1,374 @@
+// Module included in the following assemblies:
+//
+// * nodes/pods/nodes-pods-secrets-store.adoc
+//
+// IMPORTANT: This file requires you to define :secrets-store-provider: before including this module.
+
+ifeval::["{secrets-store-provider}" == "AWS Secrets Manager"]
+:aws-secrets-manager:
+[id="secrets-store-aws-secrets-manager_{context}"]
+endif::[]
+ifeval::["{secrets-store-provider}" == "AWS Systems Manager Parameter Store"]
+:aws-systems-manager-parameter-store:
+[id="secrets-store-aws-systems-manager-parameter-store_{context}"]
+// TODO: Is this the best way to do unique context for shared module in same assembly?
+endif::[]
+
+:_content-type: PROCEDURE
+= Mounting secrets from {secrets-store-provider}
+
+You can use the {secrets-store-operator} to mount secrets from {secrets-store-provider} to a CSI volume in {product-title}. To mount secrets from {secrets-store-provider}, your cluster must be installed on AWS and use AWS Security Token Service (STS).
+
+[IMPORTANT]
+====
+It is not supported to use the {secrets-store-operator} with {secrets-store-provider} in a hosted control plane cluster.
```

Comment:
@bergerhoffer The PR's AWS provider part content is all LGTM now. Go ahead with the PR's other parts with Rohit / others.
BTW minor question, whether do we need link the bug link directly?

Reply:
@xingxingxia Thanks! We don't normally put bug links into the main docs content. The bug links for known issues/fixed issues are in the release notes.

If we do fix this bug and this limitation is no longer applicable, then we can come back and remove this note.